### PR TITLE
Chromium and Safari don't support  `Worker` in Shared Workers

### PR DESCRIPTION
**Pull request by skyclouds2001**
#### Summary

the `Worker` interface is not exposed to Shared Worker for chromium-based browsers, and likely will not be fixed recently, so it make sense to add notes to indicating it is not exposed to Shared Worker

Note: I'm not sure if partial_implementation or impl_url key should be added

see https://github.com/mdn/content/pull/30880#discussion_r1422009358 and https://issues.chromium.org/issues/40695450, also https://github.com/chromium/chromium/blob/main/third_party/blink/renderer/core/workers/worker.idl

can be test using https://worker-playground.glitch.me/

> Another note: BCD for Worker interface and properties/methods and Worker constructor for chrome is different, maybe a mistake?

### CODE REVIEW

File: api/Worker.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
         "spec_url": "https://html.spec.whatwg.org/multipage/workers.html#dedicated-workers-and-the-worker-interface",
         "support": {
           "chrome": {
-            "version_added": "2"
+            "version_added": "2",
+            "notes": "Not available in Shared Workers. See <a href='https://crbug.com/40695450'>bug 40695450</a>."
```
Comment (caugner) [inline api/Worker.json:10]: Does the spec indicate that `Worker` should be available in Shared Workers? I can see that there is a separate `SharedWorker` interface.

Can you explain what it is that say Firefox supports here but Chrome doesn't?

**Comment by skyclouds2001** [comment]
researches

---

chromium

* add support for dedicated worker

commit https://github.com/chromium/chromium/commit/304237d65a5dbeaca92a2a9a276b4cb903b5ad14
chrome feature entry https://chromestatus.com/feature/6080438103703552
issue https://issues.chromium.org/issues/40340498
code review https://chromium-review.googlesource.com/c/chromium/src/+/953746

local cli test

image

---

firefox

* expose to dedicated worker & shared worker

commit https://github.com/mozilla/gecko-dev/commit/8b59c15d45ed63d4fab0482f3d5b31ef93bda274
issue https://bugzilla.mozilla.org/show_bug.cgi?id=1137398

* expose to worker

commit https://github.com/mozilla/gecko-dev/commit/12b78b96120504ec21e462eb8c99709b075cf965
issue https://bugzilla.mozilla.org/show_bug.cgi?id=1017988

note that service worker only supported via v44, so *expose to worker* is equal to *expose to dedicated worker & shared worker* between v34 to v39

---

safari

commit https://github.com/WebKit/WebKit/commit/48880e342359f100878b1b87373e706db7dfe540
issue https://bugs.webkit.org/show_bug.cgi?id=22723
browser release note https://developer.apple.com/documentation/safari-release-notes/safari-16_4-release-notes

image

File: api/Worker.json (hunk 1 above)
Comment (caugner) [inline api/Worker.json:10]: Let's remove this note, as it duplicates the information below:

```suggestion
            "version_added": "2"
```

**Review by caugner — CHANGES_REQUESTED** [review]
Overall LGTM, just one nit.

**Pull request merged**